includes/utils/class-amp-dom-utils.php

```diff
@@ -711,4 +719,174 @@ public static function recursive_force_closing_tags( $dom, $node = null ) {
 	private static function is_self_closing_tag( $tag ) {
 		return in_array( strtolower( $tag ), self::$self_closing_tags, true );
 	}
+
+	/**
+	 * Check whether a given element has a specific class.
+	 *
+	 * @since 1.4.0
+	 *
+	 * @param DOMElement $element Element to check the classes of.
+	 * @param string     $class   Class to check for.
+	 * @return bool Whether the element has the requested class.
+	 */
+	public static function has_class( DOMElement $element, $class ) {
+		if ( ! $element->hasAttribute( 'class' ) ) {
+			return false;
+		}
+
+		$classes = $element->getAttribute( 'class' );
+
+		return (bool) preg_match( "/\b{$class}\b/", $classes );
```

Comment:
Still needs to be made more robust since `\b` matches in hyphens. This should do it:

```suggestion
		return in_array( $class, preg_split( '/\s/', $classes ), true );
```